File: stack/20.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
       Stack<Character> st = new Stack<>();
       for(char c :s.toCharArray()){
        if(c == '(' || c == '{' || c == '['){
            st.push(c);
        }
        else {
            if(st.isEmpty()) return false;
            int top = st.pop();
            if((c ==')' && top!='(')||(c =='}' && top!='{')||(c ==']' && top!='[')) return false;
        }
       }
       return st.isEmpty();
        
    }
}

File: 20.cpp
class Solution {
  public:
    bool isValid(string s) {
      stack<char> st;
      unordered_map<char,char> map= {{'(',')'},{'{','}'},{'[',']'}};
      for(char c : s){
        if(c=='(' || c=='[' || c == '{'){
          st.push(c);
        }
        else {
            if(st.empty()) return false;
            if(map[st.top()] == c){
                st.pop();
            }
            else {
                return false;
            }
        }
      }
      return st.empty();
    }
};

